tests/unit/test_charm.py

```diff
@@ -2,37 +2,17 @@
 # See LICENSE file for licensing details.
 
 import unittest
-from typing import List, Optional
-from unittest.mock import patch
+from unittest.mock import MagicMock, patch
 
 import hcl
 import ops
 import ops.testing
 from charm import VaultOperatorCharm, config_file_content_matches
-from charms.operator_libs_linux.v2.snap import SnapState
+from charms.operator_libs_linux.v2.snap import Snap, SnapState
 
 PEER_RELATION_NAME = "vault-peers"
 
 
-class MockSnapObject:
```

Comment:
While I agree with this change, it does not relate to the change claimed in the PR title (i.e. commit message) or PR description.

Reply:
Well, maybe not directly, but I had to update the tests to work with the new call to `vault.latest`, and rather than continue to extend the bespoke mock, it was simpler to just switch it to a `MagicMock`. 

I didn't feel the quick refactor needed a separate PR since this was already pretty small, but I can break it out if you'd like?

Reply:
I added a bit to the description about this part. Maybe that is sufficient?